doc/source/data/transforming-datasets.rst

```diff
@@ -78,34 +80,338 @@ express your customized business logic in transformations. Here we will focus on
 :meth:`.map_batches() <ray.data.Dataset.map_batches>` as it's the primary mapping
 API in Datasets.
 
-A UDF can be a function or a callable class, which has the following input/output:
+A batch UDF can be a function or, if using the actor compute strategy, a callable class.
+These UDFs have several batch format options, which control the format of the
+batches that are passed to the provided batch UDF. Depending on the underlying dataset
+format, using a particular batch format may or may not incur a data conversion cost
+(e.g. convertion an Arrow Table to a Pandas DataFrame, or creating an Arrow Table from a
```

Comment:
If doable, it will be best to put this into a table, like for a <from, to> pair, what are the implications of conversion cost.